relay/relay.go

```diff
@@ -99,3 +99,15 @@ type Peer struct {
 	// to, which is also useful for stats.
 	Zone string
 }
+
+// RateLimitDropError is an error that a Relayer will get when it should drop
+// requests due to hitting a rate limit. This is kind of a hack, because rate
+// limiting of this nature isn't part of the actual TChannel protocol.
+//
+// When the Relayer gets an error like this when calling .Get() on an outbound
+// it will record that it has dropped the packet, but *won't* notify the client.
+type RateLimitDropError struct{}
```

Comment:
A good discussion of errors here:
http://dave.cheney.net/2016/04/27/dont-just-check-errors-handle-them-gracefully

An interface allows Muttley to have different errors (maybe there's different reasons for the error, one because of hitting some service specific limit, one because it hit a global Muttley limit), and they implement the same interface. tchannel doesn't need to know the different reasons or what the error message for each of those should be, but it just checks whether the returned error satisfied the interface